### ISSUE TITLE
Generation of Task-Complexity

### ISSUE THREAD

**Issue by ian-coccimiglio** [haesleinhuepf/human-eval-bia#77]

**Comment by haesleinhuepf** [comment]
Hi @ian-coccimiglio ,

these are great suggestions! One thing we could immediatly take over from you work: The list of tasks in this [Notebook / Figure](https://github.com/haesleinhuepf/human-eval-bia/blob/main/demo/summarize_by_case.ipynb) could be sorted by average pass rate (as you show in your first figure).

I also like the idea of grouping the tasks. In some not-published work I asked an LLM to group the tasks in categories (segmentation, denoising, tabular data wrangling, complex workflows, etc.). From this experiment I learned two things: 1) individual test-cases may appear in multiple categories. 2) As our number of test-cases is quite small it is hard to conclude something from that. I mean: to conclude something that is helpful for aN LLM user...

Do you have any ideas to rephrase your analysis in some way like: "We observe higher pass-rate in case of ???. In the opposite case of ???, we recommend to review generated code more carefully."

Also I find myself often explaining to LLM users: "Ask with a simple question. Do not go for complex code at the very beginning." Is this some advice our data could supports?

Thanks again for working on this! It's amazing to see how others dive into this data :-)

**Comment by ian-coccimiglio** [comment]
> these are great suggestions! One thing we could immediatly take over from you work: The list of tasks in this [Notebook / Figure](https://github.com/haesleinhuepf/human-eval-bia/blob/main/demo/summarize_by_case.ipynb) could be sorted by average pass rate (as you show in your first figure).

Yup, I think this ended up looking great, and is pretty informative too. I'll send a PR for this part

![performance_per_task](https://github.com/haesleinhuepf/human-eval-bia/assets/62724583/30525d72-ff48-42f5-b56e-86c7e6f898f8)

**Comment by ian-coccimiglio** [comment]
> Do you have any ideas to rephrase your analysis in some way like: "We observe higher pass-rate in case of ???. In the opposite case of ???, we recommend to review generated code more carefully."
> 
> Also I find myself often explaining to LLM users: "Ask with a simple question. Do not go for complex code at the very beginning." Is this some advice our data could supports?

Yeah, that's where I'd like this analysis to go. I think this analysis does support that advice, but I think specifically it means that LLMs have a higher success rate at solving problems with a limited scope. We can somewhat see from the above analyses that the problems that LLMs struggle the most with are larger problems which require lots of very specific and correct code and interacting libraries. But each individual piece of that problem may be independently solvable (but this may require some more analysis).

**Comment by ian-coccimiglio** [comment]
@jkh1 I agree, that is somewhat where I was going with "lines of code". In fact, originally I was going to quantify the number of functions used, but then I realized that the majority of common tasks bundle together functionality, and an LLM may just have to determine the right one.

**Comment by jkh1** [comment]
Also relevant could be the way the task is formulated as pointed out by Tischi in #79
